### PR TITLE
Add server pre-commit hook instructions

### PR DESCRIPTION
**Pull request by eliykat**
## Objective

Add instructions on how to set up the `dotnet format` pre-commit hook for server. It's in our [README.md](https://github.com/bitwarden/server#dotnet-format) but not in these docs.

### CODE REVIEW

**Comment by eliykat** [comment]
No deployment? 🤔

File: docs/getting-started/server/guide.mdx [hunk 1]
```diff
@@ -35,14 +35,20 @@ Before you start: make sure you’ve installed the recommended
...
+## Configure Git
 
-Configure Git to ignore the Prettier revision:
+1. Configure Git to ignore the Prettier revision:
 
 ```bash
 git config blame.ignoreRevsFile .git-blame-ignore-revs
 ```
 
+2. Set up the pre-commit `dotnet format` hook:
```
Comment (Hinton) [inline docs/getting-started/server/guide.mdx:46]: Maybe we should mark it as optional and add a disclaimer that dotnet format requires a full build which takes almost a minute to build.

**Review by Hinton — DISMISSED** [review]

File: docs/getting-started/server/guide.mdx (hunk 1 above)
Comment (eliykat) [reply docs/getting-started/server/guide.mdx]: good point, done

**Review by Hinton — APPROVED** [review]
LGTM thanks.

**Pull request merged**